src/System.IO/tests/Stream/Stream.ReadOnly.cs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    // This class emulates the behavior of Stream.Null,
+    // except CanWrite == false and attempting to write
+    // will throw a InvalidOperationException.
+    public sealed class ReadOnlyStream : Stream
```

Comment:
Do you need to create a new stream type?  I only see it being used once, in a test that just wants to see if an exception from its WriteAsync propagates out of a CopyToAsync call, and for that you could use any Stream that throws from its Write method, such as a ```new MemoryStream(new byte[1], writable: false)```.